### ISSUE TITLE
Using Recovered Config File Result In Empty File System

### ISSUE THREAD

**Issue by hizzely** [bailey27/cppcryptfs#147]
Hi bailey27, thanks for this wonderful project!
cppcryptfs fits my use case better than other similar tools. But before I'm putting my files, I wanna to try if I could recover the encrypted files in case something bad happens with the config file.

So I put some random files inside, unmount it, then move just the config file from the root file system. I tried to recover it by following the [provided steps](https://github.com/bailey27/cppcryptfs#recovery-of-lost-or-corrupted-config-file). But when I mounted the file system again using the new config file, it's empty.

Here's the exact steps I did, all of these steps were run without administrator privileges:
1. After putting some files, I unmounted the file system.
2. Get the master key using the command: `cppcryptfsctl.exe --printmasterkey <path to the file system>`.
3. Move just the config file (gocryptfs.conf) out of the root file system.
4. Create the second file system for recovery with the same configuration as the first one except for the path and password.
5. Run the command `cppcryptfsctl.exe --recover <path to the second file system>` then entered the first file system's master key and new password that is different to the first and second file system.
6. Mount the first file system with the new config file from second file system.
7. Successfully mounted, but empty.

Inside the second file system folder, there appears to be two config files: `gocryptfs.conf` and `gocryptfs.conf.bak`. But their content is exactly the same. So I guess the recover command has no effect? did I missed something here? I'm on version 1.4.3.11.

**Comment by bailey27** [comment]
Hi. 

I followed these steps, and I was able to recover a file system using the printed master key.

It tried again with paths with spaces in them in case I messed that up, but that also worked.

I'm not sure what's going wrong in your test.

**Comment by hizzely** [comment]
Hey thanks for the response! So I tried again and now I figured it out why it didn't work.
It's because I specified the volume name when creating the second file system used for recovery.

Somehow the program didn't give me this message if I specified the volume name:
```
key encrypted with new password written to .\gocryptfs.conf
after mounting and testing, please delete or move the backup file .\gocryptfs.conf.bak out of the way.
```

So my terminal looked like this when it didn't work:
```
PS D:\TestRecover> cppcryptfsctl.exe --recover .
Enter/paste master key all on one line, with or without dashes.
Master Key:
New Password:
Repeat:
PS D:\TestRecover>
```

This is the exit code I got: `-1073741819`.
Something in between  [this code](https://github.com/bailey27/cppcryptfs/blob/master/cppcryptfsctl/cppcryptfsctl.cpp#L434-L447) might caused the program to exit before printing the message, I presume?

I'm afraid I can't help much to pinpoint the issue. I don't have Visual Studio installed or any C++ dev tools to help with debugging right now.

**Comment by bailey27** [comment]
Yes, write_updated_config_file() is not handling the case where there is a volume name in the recovery config file and this leads to a failed recovery. 

I think the best solution is to print a warning that the volume name from the recovery config file is being ignored and ignore it. 

Otherwise I'd have to prompt for the password for the recovery config file and decrypt the master key from it and decrypt the volume name.  I think this isn't worth it.  The user can set the volume name after the filesystem is recovered.  The volume name can be set from File Explorer while the fs is mounted and is saved in the config file when it is unmounted.

**Comment by bailey27** [comment]
This was fixed in 1.4.3.12.

**Comment by hizzely** [comment]
Thanks for the update!
I agree that ignoring the volume names might be the best solution, since name would be the last thing to worry about when doing recovery.
